libr/flag/flag.c

```diff
@@ -502,6 +502,36 @@ R_API RFlagItem *r_flag_get_i(RFlag *f, ut64 off) {
 	return list ? evalFlag (f, r_list_get_top (list)) : NULL;
 }
 
+/* return the first flag that matches an offset ordered by the order of
+ * operands to the function.
+ * XXX this is kind of super ugly! */
+R_API RFlagItem *r_flag_get_by_spaces(RFlag *f, ut64 off, int nspaces, ...) {
```

Comment:
I don't know if it would be premature optimization, but I think we can make it much more efficient since it is basically a "compute max of a list of elements", where "max" however is not just a number but depends on the flagspace.

```
htspace = ht_uu_new0 ();
for i..nspaces
   space = r_flag_space_get(f, spacename);
   ht_uu_insert (htspace, (ut64)(uintptr_t)(space), i)

list = r_flag_get_list (f, off);
max_fi = Null
max = -1
for fi in list:
   ut64 pos = ht_uu_find (htspace, (ut64)(uintptr_t)fi->space, NULL)
   if pos > max:
       max = pos
       max_fi = fi

ht_uu_free (htspace)
return max_fi
```